renderdoc/driver/gl/gl_hooks_egl.cpp

```diff
@@ -1049,86 +409,45 @@ __attribute__((visibility("default"))) __eglMustCastToProperFunctionPointerType
   if(realFunc == NULL)
     return realFunc;
 
-  DLLExportHooks();
-  HookCheckGLExtensions();
-
-  // at the moment the unsupported functions are all lowercase (as their name is generated from the
-  // typedef name).
-  string lowername = strlower(string(func));
-
-  CheckUnsupported();
-
-  // for any other function, if it's not a core or extension function we know about,
-  // just return NULL
-  return NULL;
+  return (__eglMustCastToProperFunctionPointerType)
+      SharedLookupFuncPtr(func, (void*)realFunc);
 }
 
 };    // extern "C"
 
-bool OpenGLHook::PopulateHooks()
+bool EGLHook::PopulateHooks()
 {
-  bool success = true;
-
-  SetupHooks(GL);
-
-  if(eglGetProcAddress_real == NULL)
-    eglGetProcAddress_real = (PFN_eglGetProcAddress)dlsym(libGLdlsymHandle, "eglGetProcAddress");
+  SetupHooks();
 
-#undef HookInit
-#define HookInit(function)                                                                   \
-  if(GL.function == NULL)                                                                    \
-  {                                                                                          \
-    GL.function = (CONCAT(function, _hooktype))dlsym(libGLdlsymHandle, STRINGIZE(function)); \
-    eglGetProcAddress((const char *)STRINGIZE(function));                                    \
-  }
-
-// cheeky
-#undef HookExtension
-#define HookExtension(funcPtrType, function) eglGetProcAddress((const char *)STRINGIZE(function))
-#undef HookExtensionAlias
-#define HookExtensionAlias(funcPtrType, function, alias)
-
-  DLLExportHooks();
-  HookCheckGLExtensions();
-
-  CheckExtensions(GL);
-
-  // see gl_emulated.cpp
-  glEmulate::EmulateUnsupportedFunctions(&GL);
-
-  if(RenderDoc::Inst().IsReplayApp())
-    glEmulate::EmulateRequiredExtensions(&GL, &GL);
-
-  return true;
+  return SharedPopulateHooks([](const char* funcName) { return (void*)eglGetProcAddress(funcName); });
 }
 
-OpenGLHook OpenGLHook::glhooks;
-
-const GLHookSet &GetRealGLFunctions()
+const GLHookSet &GetRealGLFunctionsEGL()
 {
-  return OpenGLHook::glhooks.GetRealGLFunctions();
+  return EGLHook::glhooks.GetRealGLFunctions();
 }
 
-Threading::CriticalSection &GetGLLock()
-{
-  return glLock;
-}
+EGLHook EGLHook::glhooks;
 
+#ifndef RENDERDOC_SUPPORT_GL // FIXME
```

Comment:
I need to find the right way to determine which driver should operate on the GLWindowingData struct. For now I will ifdef out the EGL code.